sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ColumnDefinition.scala

```diff
@@ -41,7 +42,9 @@ case class ColumnDefinition(
     comment: Option[String] = None,
     defaultValue: Option[DefaultValueExpression] = None,
     generationExpression: Option[String] = None,
+    identityColumnSpec: Option[IdentityColumnSpec] = None,
     metadata: Metadata = Metadata.empty) extends Expression with Unevaluable {
+  assert(generationExpression.isEmpty || identityColumnSpec.isEmpty)
```

Comment:
This assertion needs a short sentence to contextualize. "A ColumnDefinition cannot contain both a generation expression and an identity column spec".